Assets/Scripts/Runtime/Scriptable Objects/FlowControl.cs

```diff
@@ -22,6 +24,12 @@ public void UpdateFlow()
             {
                 CheckForActivation(receiver);
             }
+
+            if (_structure.Sources.Count() > 0 && AllSourcesLnked(_structure.Sources.First()))
```

Comment:
No reason to start from the first source. Can be any segment. They all have to connect.